### PR TITLE
fix DPS bugs and resolve test issue

### LINKED ISSUE #5213: Some DPS tests won't re-record
The following DPS tests fail when run live on dev and thus cannot be re-recorded on the Knack branch.

- [ ] test_dps_lifecycle

It is failing due to some service side intermident issue, and generic wait might need to be introduced to address it.
These tests will be marked LiveScenarioTest to faciliate progress on the Knack branch.

### PR DESCRIPTION
**Pull request by YingXue**
1. Change iot dps access policy CRUD commands and linked-hub CRUD commands to be long running operations
2. Fix #5213 - Some DPS tests won't re-record

---
This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [x] Each command and parameter has a meaningful description.
- [x] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

**Comment by promptws** [comment]
View a preview at https://prompt.ws/r/Azure/azure-cli/5469
This is an **experimental preview** for @microsoft users.

File: src/command_modules/azure-cli-iot/HISTORY.rst [hunk 1]
```diff
@@ -3,6 +3,11 @@
 Release History
 ===============
 
+0.1.17
+++++++
+* Change iot dps access policy CRUD commands and linked-hub CRUD commands to be long running operations
+* Fix bug 5213 - Some DPS tests won't re-record
```
Comment (tjprescott) [inline src/command_modules/azure-cli-iot/HISTORY.rst]: This isn't a customer-facing change, so you should remove it.

File: src/command_modules/azure-cli-iot/HISTORY.rst [hunk 2]
```diff
@@ -3,6 +3,11 @@
 Release History
 ===============
 
+0.1.17
+++++++
+* Change iot dps access policy CRUD commands and linked-hub CRUD commands to be long running operations
```
Comment (tjprescott) [inline src/command_modules/azure-cli-iot/HISTORY.rst]: Recommend: 
`iot dps access policy`: Certain commands are now long-running operations.
`iot dps linked-hub`: Certain commands are now long-running operations.

File: src/command_modules/azure-cli-iot/azure/cli/command_modules/iot/commands.py [hunk 3]
```diff
@@ -14,6 +14,18 @@ def __call__(self, poller):
         return result.properties.authorization_policies
 
 
+class LinkedHubUpdateResultTransform(LongRunningOperation):  # pylint: disable=too-few-public-methods
+    def __call__(self, poller):
+        result = super(LinkedHubUpdateResultTransform, self).__call__(poller)
+        return result.properties.iot_hubs
```
Comment (tjprescott) [inline src/command_modules/azure-cli-iot/azure/cli/command_modules/iot/commands.py]: This isn't the correct use of a transform. If your commands now return a poller, the CLI will automatically do the polling for you and return the object.  Transform is used if you needed to change the structure of the returned object.  Transform callables should accept a parameter called `result` (typically) that is the object being returned.

File: src/command_modules/azure-cli-iot/HISTORY.rst [hunk 4]
```diff
@@ -5,6 +5,9 @@ Release History
 
 0.1.17
 ++++++
+* iot dps access policy: Certain commands are now long-running operations.
+* iot dps linked-hub: Certain commands are now long-running operations.
```
Comment (tjprescott) [inline src/command_modules/azure-cli-iot/HISTORY.rst]: These were already LRO. From our discussion it seems like the text should be something like:

"* `iot dps create/update`: These commands now return the credentials upon completion."

File: src/command_modules/azure-cli-iot/azure/cli/command_modules/iot/custom.py [hunk 5]
```diff
@@ -226,7 +241,9 @@ def iot_dps_linked_hub_delete(client, dps_name, resource_group_name, linked_hub)
     dps_property = IotDpsPropertiesDescription(None, None, updated_hub, dps.properties.allocation_policy, dps.properties.authorization_policies)
     dps_description = ProvisioningServiceDescription(dps.location, dps_property, dps.sku)
 
-    client.iot_dps_resource.create_or_update(resource_group_name, dps_name, dps_description)
+    if no_wait:
+        return client.iot_dps_resource.create_or_update(resource_group_name, dps_name, dps_description)
+    LongRunningOperation(cmd.cli_ctx)(client.iot_dps_resource.create_or_update(resource_group_name, dps_name, dps_description))
     return iot_dps_linked_hub_list(client, dps_name, resource_group_name)
```
Comment (tjprescott) [inline src/command_modules/azure-cli-iot/azure/cli/command_modules/iot/custom.py]: FYI, delete commands in the CLI typically return nothing. However, is this is the behavior you want, that's totally fine.

File: src/command_modules/azure-cli-iot/HISTORY.rst (hunk 4 above)
Comment (YingXue) [reply src/command_modules/azure-cli-iot/HISTORY.rst]: These commands would return credentials before I changed them but often return wrong credentials as I didn't wait for the create_and_update operations. So I guess, it is still changed to long-running? I will add --no-wait support for description.

File: src/command_modules/azure-cli-iot/azure/cli/command_modules/iot/custom.py (hunk 5 above)
Comment (YingXue) [reply src/command_modules/azure-cli-iot/azure/cli/command_modules/iot/custom.py]: I am thinking still return the list so the user could verify the right entry indeed got deleted. So I think I am keeping it :)

**Pull request merged**